src/compiler/checker.ts

```diff
@@ -4456,7 +4461,7 @@ namespace ts {
                 for (const key in props) {
                     const prop = props[key];
                     // We need to filter out partial properties in union types
-                    if (!(prop.flags & SymbolFlags.SyntheticProperty && (<TransientSymbol>prop).isPartial)) {
+                    if (!(prop.flags & SymbolFlags.Synthesized && (<TransientSymbol>prop).isPartial)) {

```

Comment:
I would just leave this flag alone, it doesn't have anything to do with the fix here.